app/Http/Controllers/CountryController.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\CountryRequest;
+use App\Http\Resources\CountryResource;
+use App\Models\Country;
+use App\Services\CountryService;
+use Inertia\Inertia;
+use Inertia\Response;
+
+class CountryController extends Controller
+{
+    public function index(CountryService $service): Response
+    {
+        $countries = $service->indexCountry();
+
+        return Inertia::render("Countries", [
+            "countries" => $countries,
+        ]);
+    }
+
+    public function store(CountryRequest $request, CountryService $service): void
+    {
+        $service->storeCountry(
+            $request->name,
+            $request->altName,
+            $request->lat,
+            $request->lon,
+            $request->iso,
+        );
+    }
+
+    public function show(Country $country, CountryService $service): CountryResource
+    {
+        return $service->showCountry($country);
+    }
+
+    public function update(CountryService $service, Country $country, CountryRequest $request): void
```

Comment:
Instead of `CountryRequest` create new class `UpdateCountryRequest` which extends `CountryRequest`.
Then in rules method you can use `arrray_merge_recursive()` php core method.

```php
use Illuminate\Validation\Rule;
use Illuminate\Validation\Rules\Unique;

public function rules(): array {
   return array_merge_recursive(
     parent::rules(),
    [
        "name" => [$this->uniqueNameForCountry()],
    ],
}

protected function uniqueNameForCountry(): Unique
    {
        $currentCountryId = $this->route(param: "country")?->id;

        return Rule::unique(table: "countires", column: "name")->ignore($currentCountryId);
    }
```

https://www.php.net/manual/en/function.array-merge-recursive.php
https://laravel.com/docs/10.x/validation#rule-unique
https://php.org/inheritance-in-php-oop/